### PR TITLE
Lambda Function URLs

### LINKED ISSUE #24050: New feature: Lambda Function URLs
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

[AWS Go SDK v1.43.34](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md#release-v14334-2022-04-06):

> This release adds new APIs for creating and managing Lambda Function URLs and adds a new FunctionUrlAuthType parameter to the AddPermission API. Customers can use Function URLs to create built-in HTTPS endpoints on their functions.

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #24050.

##### TODO

* [x] Review/edit documentation after looking at [AWS documentation](https://docs.aws.amazon.com/lambda/latest/dg/lambda-urls.html)

#### AWS Commercial

```console
% make testacc TESTS=TestAccLambdaFunctionURL PKG=lambda ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lambda/... -v -count 1 -parallel 2 -run='TestAccLambdaFunctionURL'  -timeout 180m
=== RUN   TestAccLambdaFunctionURLDataSource_basic
=== PAUSE TestAccLambdaFunctionURLDataSource_basic
=== RUN   TestAccLambdaFunctionURL_basic
=== PAUSE TestAccLambdaFunctionURL_basic
=== RUN   TestAccLambdaFunctionURL_Cors
=== PAUSE TestAccLambdaFunctionURL_Cors
=== RUN   TestAccLambdaFunctionURL_Alias
=== PAUSE TestAccLambdaFunctionURL_Alias
=== CONT  TestAccLambdaFunctionURLDataSource_basic
=== CONT  TestAccLambdaFunctionURL_Cors
--- PASS: TestAccLambdaFunctionURLDataSource_basic (53.51s)
=== CONT  TestAccLambdaFunctionURL_basic
--- PASS: TestAccLambdaFunctionURL_Cors (67.94s)
=== CONT  TestAccLambdaFunctionURL_Alias
--- PASS: TestAccLambdaFunctionURL_basic (38.30s)
--- PASS: TestAccLambdaFunctionURL_Alias (45.45s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/lambda	123.047s
```

#### AWS GovCloud

```console
% make testacc TESTS=TestAccLambdaFunctionURL PKG=lambda ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lambda/... -v -count 1 -parallel 2 -run='TestAccLambdaFunctionURL'  -timeout 180m
=== RUN   TestAccLambdaFunctionURLDataSource_basic
=== PAUSE TestAccLambdaFunctionURLDataSource_basic
=== RUN   TestAccLambdaFunctionURL_basic
=== PAUSE TestAccLambdaFunctionURL_basic
=== RUN   TestAccLambdaFunctionURL_Cors
=== PAUSE TestAccLambdaFunctionURL_Cors
=== RUN   TestAccLambdaFunctionURL_Alias
=== PAUSE TestAccLambdaFunctionURL_Alias
=== CONT  TestAccLambdaFunctionURLDataSource_basic
=== CONT  TestAccLambdaFunctionURL_Cors
=== CONT  TestAccLambdaFunctionURLDataSource_basic
    function_url_test.go:21: Lambda Function URLs are not supported in aws-us-gov partition
--- SKIP: TestAccLambdaFunctionURLDataSource_basic (1.84s)
=== CONT  TestAccLambdaFunctionURL_Cors
    function_url_test.go:21: Lambda Function URLs are not supported in aws-us-gov partition
=== CONT  TestAccLambdaFunctionURL_Alias
--- SKIP: TestAccLambdaFunctionURL_Cors (1.84s)
=== CONT  TestAccLambdaFunctionURL_basic
=== CONT  TestAccLambdaFunctionURL_Alias
    function_url_test.go:21: Lambda Function URLs are not supported in aws-us-gov partition
=== CONT  TestAccLambdaFunctionURL_basic
    function_url_test.go:21: Lambda Function URLs are not supported in aws-us-gov partition
--- SKIP: TestAccLambdaFunctionURL_Alias (0.00s)
--- SKIP: TestAccLambdaFunctionURL_basic (0.00s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/lambda	9.231s
```

### CODE REVIEW

File: website/docs/r/lambda_function_url.html.markdown [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    max_age           = 86400
+  }
+}
+```
+
+## Argument Reference
+
+* `function_name` - (Required) Lambda Function name or ARN.
+* `qualifier` - (Optional) The Lambda alias name or '$LATEST'.
+* `authorization_type` - (Required) The authorization type for your function URL. Valid values are `["AWS_IAM"]` and `["NONE"]`.
```
Comment (antonbabenko) [inline website/docs/r/lambda_function_url.html.markdown:42]:
```suggestion
* `authorization_type` - (Required) The authorization type for your function URL. Valid values are `"AWS_IAM"` and `"NONE"`.
```

**Review by antonbabenko — APPROVED** [review]

**Comment by ewbankkit** [comment]
@bnusunny Thanks for the contribution :tada: :clap:.

**Comment by guanzo** [comment]
From announcement to PR to release in a single day. Amazing. Helps that weekly releases are on Thursday too!

**Comment by bnusunny** [comment]
@guanzo  I couldn't reproduce your issue. Could you please open an issue if you still see this problem?

**Comment by guanzo** [comment]
@bnusunny I manually deleted the statement, after that I couldn't reproduce anymore. 🤷

**Pull request merged**